streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -425,7 +425,7 @@
     /**
      * Combine values of this stream with another {@link KStream}'s elements of the same key using windowed Outer Join.
      * If a record key is null it will not included in the resulting {@link KStream}
-     * Both of the joining {@link KStream}s will be materialized in local state stores with the given store names.
+     * Both of the joining {@link KStream}s will be materialized in local state stores with with auto-generated name ${storeName}.
```

Comment:
typo: ..with with..

same typo also in the other javadoc comments below.
Also, where does ${storeName} come from? Would it be enough just to say "with an auto-generated store name" ?